packages/amplify-cli/src/extensions/amplify-helpers/remove-env-from-cloud.js

```diff
@@ -0,0 +1,35 @@
+const { getProjectConfig } = require('./get-project-config');
+const { getCategoryPlugins } = require('./get-category-plugins');
+const { getProviderPlugins } = require('./get-provider-plugins');
+
+async function removeEnvFromCloud(context, envName) {
+  const { providers } = getProjectConfig();
+  const providerPlugins = getProviderPlugins(context);
+  const providerPromises = [];
+  context.print.info('');
+  context.print.info(`Deleting env:${envName}`);
+
+  providers.forEach((providerName) => {
+    const pluginModule = require(providerPlugins[providerName]);
+    providerPromises.push(pluginModule.deleteEnv(context, envName));
+  });
+
+  try {
+    await Promise.all(providerPromises);
+  } catch (e) {
+    context.print.info('');
+    context.print.error(`Error in deleting env:${envName}`);
+    context.print.info(e.message);
+    throw e;
+  }
+
+  const categoryPlugins = getCategoryPlugins(context);
+  if (categoryPlugins.notifications) {
+    const notificationsModule = require(categoryPlugins.notifications);
+    await notificationsModule.deletePinpointApp(context);
+  }
+}
+
```

Comment:
The logic for notifications deletePinpointApp is incorrect. 
Currently, that function will search the amplifyMeta and find the pinpoint App and delete it. 
For env specific delete of a pinpoint App, the pinpoint app should be searched in the team-provider-info.json. 
A new function for the notifications should be implemented and invoked here for env specific app delete.

Reply:
As discussed offline, I'll comment this out and create a stub out there and @UnleashedMind would take over the implementation of it.